trace_viewer/extras/importer/trace_event_importer.html

```diff
@@ -406,6 +409,22 @@
       this.model_.samples.push(sample);
     },
 
+    processMemoryDumpEvent: function(event) {
+      if (event.id === undefined) {
+        this.model_.importWarning({
+          type: 'memory_dump_parse_error',
+          message: 'd phase event without a dump ID.'
+        });
+        return;
+      }
+
+      if (!(event.id in this.allMemoryDumpEvents_)) {
+        this.allMemoryDumpEvents_[event.id] = [];
```

Comment:
I suggest if(this.allMemoryDumpEvents_[event.id] === undefined) ...

On Mon Feb 09 2015 at 6:18:19 PM petrcermak <notifications@github.com>
wrote:

> In trace_viewer/extras/importer/trace_event_importer.html
> <https://github.com/google/trace-viewer/pull/749#discussion_r24384554>:
>
> > @@ -406,6 +409,22 @@
> >        this.model_.samples.push(sample);
> >      },
> >
> > +    processMemoryDumpEvent: function(event) {
> > +      if (event.id === undefined) {
> > +        this.model_.importWarning({
> > +          type: 'memory_dump_parse_error',
> > +          message: 'd phase event without a dump ID.'
> > +        });
> > +        return;
> > +      }
> > +
> > +      if (!(event.id in this.allMemoryDumpEvents_)) {
> > +        this.allMemoryDumpEvents_[event.id] = [];
>
> Unfortunately, that operator throws a SyntaxError.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/trace-viewer/pull/749/files#r24384554>.
>